### PR TITLE
Adding Core Analytics Events

### PR DESCRIPTION
**Pull request by imann24**
https://app.asana.com/0/319039773705935/290183434167033

**Scripts**
- Adding classes to represent dog events, play session, and currency events
- Adding constant values to `PPGlobal` and `Global`
- Setting Mixpanel super properties in `MixpanelAnalyticsInterchange` script
- Added class extension for `TimeSpan` to print out a human readable string of time played

### CODE REVIEW

File: Assets/Plugins/Mixpanel/Mixpanel.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public static class Mixpanel
 
 	private const string API_URL_FORMAT = "https://api.mixpanel.com/track/?data={0}";
 	private static MonoBehaviour _coroutineObject;
-	private static bool sendInEditor = false;
+    private static bool sendInEditor = false;
 
 	public static void ToggleSendInEditor(bool isEnabled)
```
Comment (gracebarsny) [inline Assets/Plugins/Mixpanel/Mixpanel.cs]: Would be nice if we could toggle this in the inspector somewhere (maybe on PPAnalyticsController?).

File: Assets/Scripts/Analytics/PickupPup/DogAnalyticsEvent.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    base(id, getPropertyKeys(includeScoutingInfo), getPropertyValues(dog, includeScoutingInfo)) 
+    {
+        // NOTHING
+    }
+
+    static string[] getPropertyKeys(bool includeScoutingInfo = false)
+    {
+        if(includeScoutingInfo)
+        {
+            return new string[]{k.NAME, k.DOG_TIMER_LENGTH};
```
Comment (imann24) [inline Assets/Scripts/Analytics/PickupPup/DogAnalyticsEvent.cs]: @gracebarsny I don't recall discussing this convention.

File: Assets/Plugins/Mixpanel/Mixpanel.cs (hunk 1 above)
Comment (imann24) [inline Assets/Plugins/Mixpanel/Mixpanel.cs]: @gracebarsny this change is done and the rename. 

However, I disagree about the inline array declaration. Don't believe we ever determined a convention and would prefer how it's currently written

Comment (gracebarsny) [inline Assets/Plugins/Mixpanel/Mixpanel.cs]: Totally fine!

**Pull request merged**